_posts/2016-10-19-puppet4-integration.md

```diff
@@ -0,0 +1,148 @@
+---
+layout: post
+title: Katello Puppet 4 integration
+date: 2016-10-19
+author: Zach Huntington-Meath
+tags:
+- katello
+- puppet
+excerpt: |
+    Puppet 4 integration is coming to Katello and we wanted to share with you how it works!
+---
+
+### Puppet 4 Integration with Katello
+
+Recently, Foreman and Puppet 4 became compatible. Now, Katello is ready to work with Puppet 4. It is available in Katello nightly and will be released in version 3.2.  In this post, we will explain how to create new Katello instances that are Puppet 4 ready, and upgrade existing systems.
+
+There are some important things to know about using Katello with Puppet 4:
+
+- Puppet 4 uses the Java Virtual Machine and requires more memory than Puppet 3 did. This means that systems acting as Katello servers with Puppet 4 need a larger amount of memory. At a minimum, we are requiring 8 GB of memory, and recommending 12GB
+
+- Puppet 4 masters can still provision hosts with Puppet 3 with [some configuration](https://docs.puppet.com/puppetserver/latest/compatibility_with_puppet_agent.html). This can help with systems that are migrating over to Puppet 4. You can start with upgrading your masters to Puppet 4 and then gradually upgrade your agents.
+
+- You can use this matrix as a quick reference of what can work together:
+
+| | Puppet 3 client | Puppet 4 client
+| :--- | :---: | :---: 
+| **Puppet 4 server** | Compatible | Compatible
+| **Puppet 3 server** | Compatible | Not compatible
+
+- Katello will by default install modules to the normal file paths for the version of Puppet you are using. 
+ * Puppet 3 will install modules to `/etc/puppet/environments`
+ * Puppet 4 will install modules to `/etc/puppetlabs/code/environments`
+
+### How to upgrade Katello servers, capsules, and hosts to use Puppet 4
+
+We have a [great video](https://youtu.be/GFNPHypFhl4?t=2545) and [excellent documentation](http://www.katello.org/docs/3.2/upgrade/puppet.html) on how to upgrade from Puppet 3 to 4.
+
+To upgrade an existing Katello instance to Puppet 4:
+
+Before starting, we recommend creating a [backup](http://www.katello.org/docs/3.2/user_guide/backup/) or a snapshot of your system.
+
+Then run:
```

Comment:
Hm, I just noticed that we're missing a step here, the user has to make the packages available by enabling the Puppet 4 repo instead of Puppet 3. 

See "Upgrading" section of http://www.katello.org/docs/3.2/upgrade/puppet.html